compose-app/src/main/java/com/mapbox/maps/extension/compose/ComposeUtils.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.maps.extension.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import com.mapbox.maps.MapSurface
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettings
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettingsInterface
+import kotlinx.coroutines.awaitCancellation
+
+
+internal suspend inline fun disposingComposition(factory: () -> Composition) {
+  val composition = factory()
+  try {
+    awaitCancellation()
+  } finally {
+    composition.dispose()
+  }
+}
+
+internal fun MapView.newComposition(
+  parent: CompositionContext,
+  content: @Composable () -> Unit
+): Composition {
+  return Composition(
+    MapApplier(this, this), parent
+  ).apply {
+    setContent(content)
+  }
+}
+
+internal fun MapSurface.newComposition(
```

Comment:
Are we going to have compose support for textureView as well. if yes, maybe good idea to add `newComposition` for textureView as well ?

Reply:
The scope of the compose extension will only cover MapView, also this PR is for prototyping the POC, we will close this PR in favour of new PRs as planned.